packages/expo-cli/src/commands/init.ts

```diff
@@ -357,24 +363,47 @@ async function installNodeDependenciesAsync(projectRoot: string, packageManager:
   }
 }
 
-export async function initGitRepoAsync(
+async function checkSkipInitializeGitTree(
   root: string,
-  flags: { silent: boolean; commit: boolean } = { silent: false, commit: true }
+  cdPath: string,
+  flags: { silent: boolean } = { silent: false }
 ) {
   // let's see if we're in a git tree
+  let insideGitTree = false;
+
   try {
-    await spawnAsync('git', ['rev-parse', '--is-inside-work-tree'], {
+    const resultPromise = await spawnAsync('git', ['rev-parse', '--is-inside-work-tree'], {
       cwd: root,
     });
-    !flags.silent && Log.log('New project is already inside of a git repo, skipping git init.');
+
+    insideGitTree = resultPromise.stdout.trim() === 'true';
   } catch (e) {
     if (e.errno === 'ENOENT') {
-      !flags.silent && Log.warn('Unable to initialize git repo. `git` not in PATH.');
+      !flags.silent && Log.warn('Unable to check if within existing git repo. `git` not in PATH.');
       return false;
     }
   }
 
-  // not in git tree, so let's init
+  let shouldSkipInitializeGitTree = false;
+
+  if (insideGitTree) {
+    Log.log(
+      `Your project will be initialized inside an existing git repository (${process.cwd()})`
```

Comment:
this won't point to the root of the git repo. you need to run `git rev-parse --show-toplevel` to get the root of the git repo